test/testcafe/framework/page-objects/SelectAuthenticatorPageObject.js

```diff
@@ -1,6 +1,16 @@
-import { Selector } from 'testcafe';
+import { Selector, userVariables } from 'testcafe';
 import BasePageObject from './BasePageObject';
 
+const AUTHENTICATOR_BUTTON_SELECTOR = '[data-se="authenticator-button"]';
+const AUTHENTICATOR_BUTTON_LABEL_SELECTOR = '[data-se="authenticator-button-label"]';
+const AUTHENTICATOR_BUTTON_ICON_SELECTOR = '[data-se="authenticator-icon"] div';
+const AUTHENTICATOR_BUTTON_CTA_BUTTON_LABEL_SELECTOR = `${AUTHENTICATOR_BUTTON_SELECTOR} [data-se="cta-button-label"]`;
+const AUTHENTICATOR_BUTTON_CTA_BUTTON_SELECTOR = `${AUTHENTICATOR_BUTTON_SELECTOR} .cta-button`;
+const AUTHENTICATOR_BUTTON_DESCRIPTION_SELECTOR = '[data-se="authenticator-button-description"]';
+const AUTHENTICATOR_BUTTON_USAGE_TEXT_SELECTOR = '[data-se="authenticator-button-usage-text"]';
+const V3_CUSTOM_OTP_BUTTON_SELECTOR = `${AUTHENTICATOR_BUTTON_SELECTOR} [data-se="custom_otp"]`;
```

Comment:
Can any of these be replaced with `getByRole('button', { name: 'button text' })`?

Reply:
One issue I experienced with this is that, in v2, the authenticator button container is just a simple div (doesnt have a role of button) but in v3 we do have a role of button for the container.

What I thought about doing was using the `getByRole('button')` and using the index to assert proper order and elements within the button (since it contains more than just a label, i.e. logo, description, link, etc), but this would definitely require me to change the test. 

For example, here is one example of how the test asserts a specific authenticator:

```
await t.expect(selectFactorPage.getFactorLabelByIndex(0)).eql('Password');
  await t.expect(await selectFactorPage.factorDescriptionExistsByIndex(0)).eql(false);
  await t.expect(selectFactorPage.getFactorIconClassByIndex(0)).contains('mfa-okta-password');
  await t.expect(await selectFactorPage.factorCustomLogoExist(0)).eql(false);
  await t.expect(selectFactorPage.getFactorSelectButtonByIndex(0)).eql('Select');
  await t.expect(selectFactorPage.getFactorSelectButtonDataSeByIndex(0)).eql('okta_password');
  ``` 
  
How much autonomy do we have to update the test?

Reply:
I ended up just adding the classes from v2 into v3 to make the spec pass without introducing new conditional statements in each function. 

The way this spec is set up, it seems almost impossible to grab these elements semantically and pass the spec while also maintaining how v2 is structured for this view. 

Wanted to get your input on this.